tests/unit/restapi/v1/test_queue.py

```diff
@@ -0,0 +1,500 @@
+    assert isinstance(response["description"], str)
+    assert isinstance(response["createdOn"], str)
+    assert isinstance(response["lastModifiedOn"], str)
+    assert isinstance(response["latestSnapshot"], bool)
+
+    assert response["name"] == expected_contents["name"]
+    assert response["description"] == expected_contents["description"]
+
+    assert helpers.is_iso_format(response["createdOn"])
+    assert helpers.is_iso_format(response["lastModifiedOn"])
+
+    # Validate the UserRef structure
+    assert isinstance(response["user"]["id"], int)
+    assert isinstance(response["user"]["username"], str)
+    assert isinstance(response["user"]["url"], str)
+    assert response["user"]["id"] == expected_contents["user_id"]
+
+    # Validate the GroupRef structure
+    assert isinstance(response["group"]["id"], int)
+    assert isinstance(response["group"]["name"], str)
+    assert isinstance(response["group"]["url"], str)
+    assert response["group"]["id"] == expected_contents["group_id"]
+
+    # Validate the TagRef structure
+    for tag in response["tags"]:
+        assert isinstance(tag["id"], int)
+        assert isinstance(tag["name"], str)
+        assert isinstance(tag["url"], str)
+
+
+def assert_retrieving_queue_by_id_works(
+    client: FlaskClient,
+    queue_id: int,
+    expected: dict[str, Any],
+) -> None:
+    """Assert that retrieving a queue by id works.
+
+    Args:
+        client: The Flask test client.
+        queue_id: The id of the queue to retrieve.
+        expected: The expected response from the API.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_QUEUES_ROUTE}/{queue_id}", follow_redirects=True
+    )
+    assert response.status_code == 200 and response.get_json() == expected
+
+
+def assert_retrieving_queues_works(
+    client: FlaskClient,
+    expected: list[dict[str, Any]],
+    group_id: int | None = None,
+    search: str | None = None,
+    paging_info: dict[str, Any] | None = None,
+) -> None:
+    """Assert that retrieving all queues works.
+
+    Args:
+        client: The Flask test client.
+        expected: The expected response from the API.
+        group_id: The group ID used in query parameters.
+        search: The search string used in query parameters.
+        paging_info: The paging information used in query parameters.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the API response
+            does not match the expected response.
+    """
+
+    query_string = {}
+    if group_id is not None:
+        query_string["groupId"] = group_id
+    if search is not None:
+        query_string["query"] = search
+    if paging_info is not None:
+        query_string["index"] = paging_info["index"]
+        query_string["pageLength"] = paging_info["page_length"]
```

Comment:
Minor whitespace nitpick, I prefer to separate `if` blocks with a blank line (I think it's more readable), take or leave:

```suggestion
    query_string = {}

    if group_id is not None:
        query_string["groupId"] = group_id

    if search is not None:
        query_string["query"] = search

    if paging_info is not None:
        query_string["index"] = paging_info["index"]
        query_string["pageLength"] = paging_info["page_length"]
```